scripts/generate-style-code.js

```diff
@@ -383,6 +383,17 @@ for (const layer of layers) {
   writeIfModified(`MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/style/${camelize(layer.type)}LayerTest.java`, layerJavaUnitTests(layer));
 }
 
+// Jni
+const layerHpp = ejs.compile(fs.readFileSync('vendor/mapbox-gl-native/platform/android/src/style/layers/layer.hpp.ejs', 'utf8'), {strict: true});
+const layerCpp = ejs.compile(fs.readFileSync('vendor/mapbox-gl-native/platform/android/src/style/layers/layer.cpp.ejs', 'utf8'), {strict: true});
+
+for (const layer of layers) {
+  const layerFileName = layer.type.replace('-', '_');
+
+  writeIfModified(`vendor/mapbox-gl-native/platform/android/src/style/layers/${layerFileName}_layer.hpp`, layerHpp(layer));
+  writeIfModified(`vendor/mapbox-gl-native/platform/android/src/style/layers/${layerFileName}_layer.cpp`, layerCpp(layer));
+}
```

Comment:
:+1: should this jni generator run on the gl-native side? I think it also make sense to add it here, because we are going to migrate the jni code to this repo. Currently we need one extra step to commit this change in the gl-native submodule. 
cc @tobrun